### ISSUE TITLE
Login command may hang for a while after authorizing through the web dashboard

### ISSUE THREAD

**Issue by roman-mazur** [balena-io/balena-cli#1404]
- **balena CLI version:** 11.8.3
- **Operating system version:** macOS 10.14.5
- **32/64 bit OS and processor:** 64
- **Install method:** npm
- **If npm install, Node.js and npm version:** Node v10.16.0, npm 6.9.2

I can reproduce this almost every time:
- `balena login`
- Select "Web authorization"
- Click on "Authorize" on the dashboard
- See a page saying authorization has been granted with a button to go to the dashboard

I expect the CLI process to exit at this moment (there is no reason why it should hang). However, in practice, it may hang up to 1 minute until it understands login has succeeded.
If you interrupt the CLI process, you can use balena CLI commands - so the token is actually delivered to the CLI, but for some the process continues to wait.

**Comment by roman-mazur** [comment]
It has been bothering for a couple of months... Every time I promised I will look into the code and find the reason, but it never was a good moment, so finally decided to record the issue (didn't find other ope ticket).

**Comment by roman-mazur** [comment]
This might be related: #455

**Comment by pdcastro** [comment]
Possibly related to https://github.com/balena-io/balena-cli/issues/1436#issuecomment-548401846 (CC: @srlowe)

**Comment by pdcastro** [comment]
Useful data point thanks to an investigation on #1765: replacing the browser with the following curl command line allows web authorization to complete "immediately", while the browser may hang for up to a full minute:

```
curl -i -X POST \
   -H "Content-Type:application/x-www-form-urlencoded" \
   -H "Accept-Language:en-US,en;q=0.9" \
   -H "Cache-Control:max-age=0" \
   -H "Connection:keep-alive" \
   -H "DNT:1" \
   -H "Host:[::1]:8989" \
   -H "Origin:null" \
   -H "Sec-Fetch-Dest:document" \
   -H "Sec-Fetch-Mode:navigate" \
   -H "Sec-Fetch-Site:cross-site" \
   -H "Sec-Fetch-User:?1" \
   -H "Upgrade-Insecure-Requests:1" \
   -H "User-Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36" \
   -H "Accept:text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9" \
   -d \
'token=[...SCRUBBED... MY TOKEN WAS HERE ...]' \
 'http://127.0.0.1:8989/auth'
```

**Comment by pdcastro** [comment]
Another interesting data point, collected just now on macOS 10.14.6:
* [Firefox 75](https://www.whatsmybrowser.org/b/11IJ4): takes about 1 second
* Chrome 80: takes about 1 minute
* [Chrome 81](https://www.whatsmybrowser.org/b/EM8US): takes about 1 minute

**Comment by pdcastro** [comment]
Fixed in CLI release v11.35.2.
